### PR TITLE
Add game logic

### PR DESCRIPTION
**Pull request by szeliga71**
game logic

### CODE REVIEW

File: src/main/java/org/example/gamePlay/GamePlay.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    private boolean wouldYouPlay = true;
+    private int remainingCards = 0;
+
+    public GamePlay(House house, Player player) {
+        this.house = house;
+        this.player = player;
+    }
+
+    public void start() {
+        String deckId = startNewGame();
```
Comment (blazejprzyluski) [inline src/main/java/org/example/gamePlay/GamePlay.java]: why is starting new game returning a deck Id?

File: src/main/java/org/example/gamePlay/GamePlay.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+            if (remainingCards < 2) {
+                deckId = ifNotEnoughCardInDeck();
+            }
+            promptForNewGame(deckId);
+        }
+    }
+
+    private String startNewGame() {
+        player.setPlayerPoints(100);
+        System.out.println(" Player otrzymuje 100 punktow ! ");
```
Comment (blazejprzyluski) [inline src/main/java/org/example/gamePlay/GamePlay.java]: gracz

File: src/main/java/org/example/gamePlay/GamePlay.java [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        wouldYouPlay = true;
```
Comment (blazejprzyluski) [inline src/main/java/org/example/gamePlay/GamePlay.java]: wouldYouPlay already set to true

File: src/main/java/org/example/gamePlay/GamePlay.java [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+        String json = deckService.getJsonFromNewDeck(validators.enterNumberOfDeck(new Scanner(System.in)));
```
Comment (blazejprzyluski) [inline src/main/java/org/example/gamePlay/GamePlay.java]: what does the json contain?

File: src/main/java/org/example/gamePlay/GamePlay.java [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+        player.setPlayerWin(false);
+    }
+
+    private void endGame() {
+        System.out.println("koniec gry ");
+        wouldYouPlay = false;
+        player.setPlayerPoints(0);
+    }
+}
+
```
Comment (blazejprzyluski) [inline src/main/java/org/example/gamePlay/GamePlay.java]: unnecessary white space

File: src/main/java/org/example/model/House.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    public void addCardToHand(Card card) {
+        hand.add(card);
+    }
+
+    public int countScore(List<Card> hand) {
+        return (int) sumValuesOfCards(hand);
+    }
+
+    private long sumValuesOfCards(List<Card> cards) {
+        return cards.stream().mapToInt(this::numberValueFromCard).sum();
```
Comment (blazejprzyluski) [inline src/main/java/org/example/model/House.java]: Unnecessary mapping to int in countScore method can be fixed here by using collect method

File: src/main/java/org/example/model/Player.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+public class Player extends House {
+
+    private final String nick;
+    private int playerPoints = 0;
+    private boolean playerWin;
+
+    public Player(String nick) {
+        this.nick = nick;
+        playerWin = false;
+        new ArrayList<>();
```
Comment (blazejprzyluski) [inline src/main/java/org/example/model/Player.java]: to be deleted?

**Review by blazejprzyluski — COMMENTED** [review]
Let's discuss tests on our meeting

File: src/main/java/org/example/gamePlay/GamePlay.java (hunk 2 above)
Comment (szeliga71) [reply src/main/java/org/example/gamePlay/GamePlay.java]: fixed

File: src/main/java/org/example/gamePlay/GamePlay.java (hunk 3 above)
Comment (szeliga71) [reply src/main/java/org/example/gamePlay/GamePlay.java]: I forgot remove, after testing part of code.Fixed.

File: src/main/java/org/example/gamePlay/GamePlay.java (hunk 4 above)
Comment (szeliga71) [reply src/main/java/org/example/gamePlay/GamePlay.java]: This json contains the response body from the getShuffleDecks method in the DeckApiHandler class. It is the source of information from which I obtain the deckId and the number of cards remaining in the deck.

File: src/main/java/org/example/gamePlay/GamePlay.java (hunk 5 above)
Comment (szeliga71) [reply src/main/java/org/example/gamePlay/GamePlay.java]: fixed

File: src/main/java/org/example/model/Player.java (hunk 7 above)
Comment (szeliga71) [reply src/main/java/org/example/model/Player.java]: I forgot to delete it. Initially, Player and House were independent classes.

Comment (szeliga71) [reply src/main/java/org/example/model/Player.java]: fixed

File: src/main/java/org/example/model/House.java (hunk 6 above)
Comment (szeliga71) [reply src/main/java/org/example/model/House.java]: I corrected it, but I don't understand why this solution is correct.

File: src/main/java/org/example/gamePlay/GamePlay.java (hunk 1 above)
Comment (szeliga71) [reply src/main/java/org/example/gamePlay/GamePlay.java]: The variable deckId is obtained after fetching a new deck. I need this variable to draw cards and to check how many cards are left in the deck.

File: src/main/java/org/example/gamePlay/GamePlay.java [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+        String deckId = startNewGame();
+        while (wouldYouPlay && player.getPlayerPoints() > 0) {
+            if (remainingCards < 2) {
+                deckId = ifNotEnoughCardInDeck();
+            }
+            promptForNewGame(deckId);
+        }
+    }
+
+    private String startNewGame() {
```
Comment (blazejprzyluski) [inline src/main/java/org/example/gamePlay/GamePlay.java]: Change the method name to reflect what it does

File: src/main/java/org/example/gamePlay/GamePlay.java [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+            String decision = validators.makeDecision(new Scanner(System.in));
+            if (decision.equals("g")) {
+                answer = handlePlayerDraw(deckId);
+            } else if (decision.equals("p")) {
+                answer = handlePlayerPass(deckId);
+            }
+        }
+    }
+
+    private String ifNotEnoughCardInDeck() {
```
Comment (blazejprzyluski) [inline src/main/java/org/example/gamePlay/GamePlay.java]: Add  remainingCards = deckService.getNumbersOfRemainigCardsFromHttpResponse(json); and use this method in startNewGame

File: src/test/java/GamePlayTest.java [hunk 10]
```diff
@@ -1,165 +1,88 @@
...
+    @Test
+    void testGameEndsWhenPlayerPointsAreZero() {
         gamePlay.start();
-
-       verify(player).setPlayerPoints(100);
-        verify(validators).enterAmountOfDeck(any(Scanner.class));
-        verify(deckService).getJsonFromNewDeck(anyInt());
-        verify(deckService).getNumbersOfRemainigCardsFromHttpResponse(anyString());
-        verify(deckService).getNewDeckAndDeckId(anyString());
+        Assertions.assertFalse(gamePlay.isWouldYouPlay());
```
Comment (blazejprzyluski) [inline src/test/java/GamePlayTest.java:58]: I don't see how this checks if the points are zero

**Review by blazejprzyluski — APPROVED** [review]

**Pull request merged**